syntax = "proto3";

package chat;

enum msg_type
{
    cmd = 0;
    login = 1;
    list_record = 2;
    chat_broadcast = 3; 
    chat_private = 4;  
}

enum msg_req
{
    request = 0;
    response = 1;
}

enum cmd_type
{
    welcome = 0;
    enter_room = 1;
    leave_room = 2;
}

enum ret_status
{
    failed = 0;
    sucess = 1;
}

message ChatMsgHead{
    msg_type type = 1;

    msg_req req = 2;

    uint64 time = 3;
    uint32 msg_len = 4;
}

message Cmd{
    uint32 uid = 1;
    cmd_type cmdtpye = 2; //命令
    ret_status status = 3;  //回复状态
}

message Login{
    uint32 uid = 1;
    string password = 2;
    string uname = 3;
    ret_status status = 4;  //回复状态
}

message ChatPrivate{
    uint32 from_id = 1;
    uint32 to_id = 2;
    string content = 3;
    uint32 content_len = 4;  //对话内容长度，校验用
}

message ChatPublic{
    uint32 from_id = 1;
    string content = 2;
    uint32 content_len = 3;  //对话内容长度，校验用
}

